specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+  /// a unique stream ID, enabling the host to stream from different sources
+  /// concurrently.
+  /// The host should call this only once for unique streamId. The second
+  /// call of already created streamId without destroying
+  /// ICoreWebView2StagingTextureStream object will return an error.
+  /// 'd3dDevice' is used for creating shared IDXGI resource and NT shared
+  /// of it. The host should use Adapter of the LUID from the GetRenderAdapterLUID
+  /// for creating the D3D Device.
+  HRESULT CreateTextureStream(
+      [in] LPCWSTR streamId,
+      [in] IUnknown* d3dDevice,
+      [out, retval ] ICoreWebView2StagingTextureStream** value);
+  /// Get the graphics adapter LUID of the renderer. The host should use this
+  /// LUID adapter when creating D3D device to use with CreateTextureStream().
+  [propget] HRESULT RenderAdapterLUID([out, retval] LUID* luid);
+  /// Listens for change of graphics adapter LUID of the browser.
+  /// The host can get the updated LUID by RenderAdapterLUID. It is expected
+  /// that the host updates texture's d3d Device with UpdateD3DDevice,
+  /// removes existing buffers and creates new buffer.
+  HRESULT add_RenderAdapterLUIDChanged(
+      [in] ICoreWebView2StagingRenderAdapterLUIDChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for start stream request.
+  HRESULT remove_RenderAdapterLUIDChanged(
+      [in] EventRegistrationToken token);
+}
+/// This is the interface that handles texture streaming.
+[uuid(afca8431-633f-4528-abfe-7fc3bedd8962), object, pointer_default(unique)]
+interface ICoreWebView2StagingTextureStream : IUnknown {
+  /// Get the stream ID of the object that is used when calling CreateTextureStream.
+  /// The caller must free the returned string with CoTaskMemFree. See
+  /// [API Conventions](/microsoft-edge/webview2/concepts/win32-api-conventions#strings).
+  // MSOWNERS: TBD (wv2core@microsoft.com)
+  [propget] HRESULT StreamId([out, retval] LPWSTR* id);
+  /// Adds an allowed url origin for the given stream id. The stream requests
+  /// could be made from any frame, including iframes, but these origins
+  /// must be registered first in order for the request to succeed.
+  /// The added filter will be persistent until
+  /// ICoreWebView2StagingTextureStream is destroyed or
+  /// RemoveAllowedOrigin is called.
+  /// The renderer does not support wildcard so it will compare
+  /// literal string input to the requesting frame origin. So, the input string
+  /// should have a scheme like https://.
+  /// For example, https://www.valid-host.com, http://www.valid-host.com are
+  /// valid origins but  www.valid-host.com, or *.valid-host.com. are not
+  /// valid origins.
+  /// getTextureStream() will fail unless the requesting frame's origin URL is
+  /// added to the request filter.
+  HRESULT AddAllowedOrigin([in] LPCWSTR origin);
+  /// Remove added origin, which was added by AddAllowedOrigin.

```

Comment:
Need to describe when the remove is applied. If we're currently streaming textures to an origin and in the middle of that we call remove does that origin get cut off right away? Or it only blocks future startRequested calls? I think either is fine, but we need to document it.

Reply:
/// The allowed or disallowed origins will take effect only when Javascript
  /// request a streaming. So, once the streaming started, it does not stop
  /// streaming.